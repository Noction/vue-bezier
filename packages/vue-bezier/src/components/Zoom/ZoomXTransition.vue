<template>
  <component
    :is="componentType"
    :tag="tag"
    enter-active-class="zoom-in-x"
    move-class="zoom-move"
    leave-active-class="zoom-out-x"
    v-bind="{ ...$attrs, ...hooks }"
  >
    <slot />
  </component>
</template>

<script lang="ts">
export const customProps = {
  styles: {
    default: () => {
      return {
        animationFillMode: 'both',
        animationTimingFunction: 'cubic-bezier(.55,0,.1,1)'
      }
    },
    type: Object
  }
}

export default {
  inheritAttrs: false
}
</script>

<script setup lang="ts">
import {
  buildComponentType,
  buildEmits,
  buildHooks,
  buildProps,
  buildTag
} from '../../composable'

const props = defineProps(buildProps(customProps))
const emit = defineEmits(buildEmits())

const componentType = buildComponentType(props)
const tag = buildTag(props)
const hooks = buildHooks(props, emit)

</script>

<style lang="scss">
  @import 'move';

  @keyframes zoom-in-x {
    0% {
      opacity: 0;
      transform: scaleX(0);
    }

    50% { opacity: 1; }
  }

  @keyframes zoom-out-x {
    0% { opacity: 1; }

    50% {
      opacity: 0;
      transform: scaleX(0);
    }

    100% { opacity: 0; }
  }

  .zoom-in-x { animation-name: zoom-in-x; }
  .zoom-out-x { animation-name: zoom-out-x; }
</style>
