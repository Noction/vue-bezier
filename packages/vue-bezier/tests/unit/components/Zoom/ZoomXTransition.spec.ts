import ZoomXTransition from '../../../../src/components/Zoom/ZoomXTransition.vue'
import { mount } from '@vue/test-utils'

describe('components / Zoom / ZoomXTransition', () => {
  it('When mounting component with default props, it should work fine', () => {
    const wrapper = mount(ZoomXTransition)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('When mounting component with group enabled prop, it should work fine', () => {
    const wrapper = mount(ZoomXTransition, { props: { group: true } })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
