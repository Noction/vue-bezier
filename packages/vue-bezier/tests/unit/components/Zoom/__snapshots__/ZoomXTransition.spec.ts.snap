// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components / Zoom / ZoomXTransition > When mounting component with default props, it should work fine 1`] = `"<transition-stub enteractiveclass=\\"zoom-in-x\\" leaveactiveclass=\\"zoom-out-x\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\" tag=\\"span\\" move-class=\\"zoom-move\\"></transition-stub>"`;

exports[`components / Zoom / ZoomXTransition > When mounting component with group enabled prop, it should work fine 1`] = `"<transition-group-stub tag=\\"span\\" enteractiveclass=\\"zoom-in-x\\" moveclass=\\"zoom-move\\" leaveactiveclass=\\"zoom-out-x\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\"></transition-group-stub>"`;
